server_src/services/database/server-settings.service.js

```diff
@@ -0,0 +1,294 @@
+
+    //Filter out the current settings from cache
+    const filter = this.currentServerSettings._id;
+
+    // Find the filter and update the server settings with the new value...
+    const updatedSettings = await ServerSettingsDB.findOneAndUpdate(
+      { _id: filter },
+      { filament: value },
+      {
+        new: true
+      }
+    );
+    // Make sure to replicate the changes in cache.
+    this.currentServerSettings["filament"] = value;
+
+    return updatedSettings["filament"];
+  }
+  /**
+   * Returns the history collection settings object
+   * @returns {Object} Object from cache for the history collection settings
+   */
+  get historyCollectionSettings() {
+    return this.currentServerSettings["history"];
+  }
+  /**
+   * Updates the polling settings for OctoPrints websocket connection in the database and then the cache.
+   * @param {Object} value containing the seconds key for OctoPrints websocket polling. String value of polling time.
+   * @throws {Error} value not provided or empty.
+   * @returns {Object} containing the new settings key
+   */
+  async updateHistoryCollectionSettings(value = {}) {
+    if (!value || isEmpty(value))
+      throw new Error("No value provided to update settings");
+
+    //Filter out the current settings from cache
+    const filter = this.currentServerSettings._id;
+
+    // Find the filter and update the server settings with the new value...
+    const updatedSettings = await ServerSettingsDB.findOneAndUpdate(
+      { _id: filter },
+      { history: value },
+      {
+        new: true
+      }
+    );
+    // Make sure to replicate the changes in cache.
+    this.currentServerSettings["history"] = value;
+
+    return updatedSettings["history"];
+  }
+  /**
+   * Returns the influx database settings object
+   * @returns {Object} Object from cache for the influx database settings
+   */
+  get influxDatabaseSettings() {
+    return this.currentServerSettings["influxExport"];
+  }
+  /**
+   * Updates the polling settings for OctoPrints websocket connection in the database and then the cache.
+   * @param {Object} value containing the seconds key for OctoPrints websocket polling. String value of polling time.
+   * @throws {Error} value not provided or empty.
+   * @returns {Object} containing the new settings key
+   */
+  async updateInfluxDatabaseSettings(value = {}) {
+    if (!value || isEmpty(value))
+      throw new Error("No value provided to update settings");
+
+    //Filter out the current settings from cache
+    const filter = this.currentServerSettings._id;
+
+    // Find the filter and update the server settings with the new value...
+    const updatedSettings = await ServerSettingsDB.findOneAndUpdate(
+      { _id: filter },
+      { influxExport: value },
+      {
+        new: true
+      }
+    );
+    // Make sure to replicate the changes in cache.
+    this.currentServerSettings["influxExport"] = value;
```

Comment:
To summarize my feedback: lot of repetition in this file, in general a good direction but it can be improved. Try to be wary when you find yourself copying code structures.

Reply:
Yeah I know, but I'm personally leaning more towards creating explicit functions to handle these inside the code, rather than having to know the database values before hand and passing them through. 

I could create a function for updating the cache, but It was extra code to me vs repetition. 

So: 
```this.currentServerSettings.influxExport = value;```
VS:
``` this.updateServerSettingsCachce("influxExport", value);```
```updateServerSettingsCahce(key, value){ this.currentServerSettings[key] = value; }```

Is there another way?